dotcom-rendering/src/components/EuropeLandingModal.importable.tsx

```diff
@@ -0,0 +1,343 @@
+
+const coe: string[] = [
+	'AL',
+	'AD',
+	'AM',
+	'AT',
+	'AZ',
+	'BE',
+	'BA',
+	'BG',
+	'HR',
+	'CY',
+	'CZ',
+	'DK',
+	'EE',
+	'FI',
+	'FR',
+	'GE',
+	'DE',
+	'GR',
+	'HU',
+	'IS',
+	'IE',
+	'IT',
+	'LV',
+	'LI',
+	'LT',
+	'LU',
+	'MT',
+	'MC',
+	'ME',
+	'NL',
+	'MK',
+	'NO',
+	'PL',
+	'PT',
+	'MD',
+	'RO',
+	'SM',
+	'RS',
+	'SK',
+	'SI',
+	'ES',
+	'SE',
+	'CH',
+	'TR',
+	'UA',
+];
+
+export const isValidEdition = guard(['EUR', 'US', 'UK', 'AU', 'INT'] as const);
+
+export const getModalType = (): ModalType => {
+	const editionCookie = getCookie({ name: 'GU_EDITION' });
+
+	const geoCountryCookie = getCookie({ name: 'GU_geo_country' });
+	const modalDismissedCookie = getCookie({ name: 'GU_eu_modal_dismissed' });
+
+	if (!geoCountryCookie) {
+		return 'NoModal';
+	}
+	if (modalDismissedCookie) {
+		return 'NoModal';
+	}
+	// If selected INT and not in COE show do u want to switch
+	if (editionCookie === 'INT' && !coe.includes(geoCountryCookie)) {
+		return 'ModalDoYouWantToSwitch';
+	}
+	// If in COE
+	if (coe.includes(geoCountryCookie)) {
+		if (editionCookie === 'INT' || !editionCookie) {
+			// If INT edition or no edition
+			return 'ModalSwitched';
+		} else {
+			// If US, UK, AUS edition
+			return 'ModalDoYouWantToSwitch';
+		}
+	}
+	return 'NoModal';
+};
+
+interface Props {
+	edition: EditionId;
+}
+export const EuropeLandingModal = ({ edition }: Props) => {
+	const editionCookie = getCookie({ name: 'GU_EDITION' });
+	const modalType = getModalType();
+	const [switchEdition, setSwitchEdition] = useState(false);
+	const [selectedEdition, setSelectedEdition] = useState<EditionId>(
+		isValidEdition(editionCookie) ? editionCookie : edition,
+	);
+
+	useOnce(() => {
+		const europeModal = document.getElementById('europe-modal-dialog');
+		if (
+			europeModal instanceof HTMLDialogElement &&
+			modalType !== 'NoModal'
+		) {
+			europeModal.showModal();
+			europeModal.addEventListener('close', () => {
+				hideModal();
+			});
+			document.documentElement.style.overflow = 'hidden';
+		}
+	}, []);
+
+	const confirmNewEdition = (editionId: EditionId) => {
+		setCookie({
+			name: 'GU_EDITION',
+			value: editionId,
+			isCrossSubdomain: true,
+		});
+		setCookie({
+			name: 'GU_eu_modal_dismissed',
+			value: 'true',
+			daysToLive: 100, //todo - check how long
+		});
+		if (editionId === edition) {
+			hideModal();
+		} else {
+			window.location.replace(getEditionFromId(editionId).url);
+		}
+	};
+
+	const dismissModal = () => {
+		setCookie({
+			name: 'GU_eu_modal_dismissed',
+			value: 'true',
+			daysToLive: 100, //todo - check how long
+		});
+		hideModal();
+	};
+
+	const hideModal = () => {
+		const europeModal = document.getElementById('europe-modal-dialog');
+		if (europeModal instanceof HTMLDialogElement) {
+			europeModal.close();
+			document.documentElement.style.overflow = '';
+		}
+	};
+
+	return (
+		<dialog css={dialogStyles} id={'europe-modal-dialog'}>
+			<div css={styles}>
```

Comment:
I'm thinking its might be possible to get rid of this internal div and just apply the styles directly to the dialog instead. The issue we were facing was trying to get the dialog to be horizontally centred, but it looks like by default we might already get that behaviour.

I tried doing a centred dialog in this codepen and it wasn't too much of a problem https://codepen.io/Ashleigh-Carr/pen/gOZLEYj